addons/website_crm_partner_assign/tests/test_partner_assign.py

```diff
@@ -43,64 +43,55 @@ def geo_find(addr, **kw):
-             'type': 'lead'},
-            {'name': 'Children 2 Lead 2',
-             'partner_id': self.customer_uk.child_ids[1].id,
-             'type': 'lead'},
-        ])
+        child_1_1 = partner.create({
+            'name': 'Child-1.1',
+            'parent_id': child_1.id,
+        })
+
         children_leads_assigned = self.env['crm.lead'].create([
             {'name': 'Children 1 Lead 1',
-             'partner_assigned_id': self.customer_uk.child_ids[0].id,
-             'type': 'lead'},
+             'partner_assigned_id': child_1.id,
+             'type': 'opportunity'},
             {'name': 'Children 1 Lead 2',
-             'partner_assigned_id': self.customer_uk.child_ids[0].id,
+             'partner_assigned_id': child_1.id,
+             'type': 'opportunity'},
+            {'name': 'Children 1.1 Lead 1',
+             'partner_assigned_id': child_1_1.id,
              'type': 'lead'},
-            {'name': 'Children 2 Lead 1',
-             'partner_assigned_id': self.customer_uk.child_ids[1].id,
-             'type': 'lead'},
-            {'name': 'Children 2 Lead 2',
-             'partner_assigned_id': self.customer_uk.child_ids[1].id,
+            {'name': 'Children 1.1 Lead 2',
+             'partner_assigned_id': child_1_1.id,
              'type': 'lead'},
         ])
-
-        self.assertEqual(
-            repr(self.customer_uk.action_view_opportunity()['domain']),
-            repr([('id', 'in', sorted(self.lead_uk.ids + lead_uk_assigned.ids + children_leads.ids))]),
-            'Parent: own + children leads + assigned'
-        )
-        self.assertEqual(
-            repr(self.customer_uk.child_ids[0].action_view_opportunity()['domain']),
-            repr([('id', 'in', sorted(children_leads[0:2].ids + children_leads_assigned[0:2].ids))]),
-            'Children: own leads + assigned'
-        )
-        self.assertEqual(
-            repr(self.customer_uk.child_ids[1].action_view_opportunity()['domain']),
-            repr([('id', 'in', sorted(children_leads[2:].ids + children_leads_assigned[2:].ids))]),
-            'Children: own leads + assigned'
-        )
+        expected_lead_count = [4, 4, 2]
+        partners = [parent, child_1, child_1_1]
+        for partner, count in zip(partners, expected_lead_count):
+            self.assertEqual(partner.opportunity_count, count, "Child partner's assign leads should propagate to parent partner.")
+
+        for lead in children_leads_assigned:
+            if lead.name == 'Children 1.1 Lead 2':
+                lead.partner_id = parent.id  # set parent as customer
+            else:
+                lead.partner_id = lead.partner_assigned_id.id  # assigning the customer same as the assigned partner
+        for partner, count in zip(partners, expected_lead_count):
+            self.assertEqual(partner.opportunity_count, count, "When both `partner_id` and `partner_assigned_id` is set as same partner count should not be changed.")
```

Comment:
Also, add a comment for the case when you add a parent as a customer.
Can't we add the comment before the case starts and remove it from here? From my POV it is better for readability.(Same for all)